### PR TITLE
Nicer `driver` formatting

### PR DESCRIPTION
**Pull request by sistemd**
Reformat the `driver` files with a `rustfmt.toml` that I really like. I think the main point here is that I use
```
imports_layout = "HorizontalVertical"
imports_granularity = "One"
```
which nicely collapses the imports so they don't look schizophrenic. It also neatly formats comments and strings.

If others like this it might not be a bad idea to start using this on all of our code.

### CODE REVIEW

**Comment by vkgnosis** [comment]
We already have a top level `rustfmt.toml`. I'd like to have consistent formatting for the whole repo so we should make changes there.

I dislike `unstable_features = true` which we already have because it makes rustfmt only work on nightly and I don't like forcing people to have nightly.

I don't have a strong preference on the formatting specifics as long as all the rules are automated by cargo. There is some value in using the defaults because it makes code look the same across projects.

Will comment on the specific options you're setting later.

**Comment by vkgnosis** [comment]
If all the settings changed in this PR are for things that are currently not covered by rustfmt at all then that point is moot, yes.

**Comment by vkgnosis** [comment]
Right now the only thing we use nightly for is the imports setting, so it would be easy to revert. If we start using other options it would be more annoying. When the PR that introduced the nightly requirement was merged I was on vacation so I didn't get to comment on that then so I wanted to bring it up here. It's a small problem that is probably trumped by better formatting.

**Comment by vkgnosis** [comment]
I noticed several options are set to their default values so they don't change anything. Why? For example fn_single_line .

File: crates/driver/rustfmt.toml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+format_strings = true
+imports_layout = "HorizontalVertical"
+imports_granularity = "One"
+normalize_doc_attributes = true
+reorder_imports = true
+reorder_impl_items = true
+group_imports = "StdExternalCrate"
+use_try_shorthand = true
+wrap_comments = true
+max_width = 100
```
Comment (vkgnosis) [inline crates/driver/rustfmt.toml:14]: already the default

File: crates/driver/rustfmt.toml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+overflow_delimited_expr = true
```
Comment (vkgnosis) [inline crates/driver/rustfmt.toml:15]: changes something that stable rustfmt already handles (but i'm fine with the change)

File: crates/driver/rustfmt.toml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+tab_spaces = 4
+edition = "2021"
+fn_single_line = false
+format_code_in_doc_comments = true
+format_strings = true
+imports_layout = "HorizontalVertical"
```
Comment (vkgnosis) [inline crates/driver/rustfmt.toml:6]: changes something that stable rustfmt already handles (but i'm fine with the change)

File: crates/driver/rustfmt.toml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+normalize_doc_attributes = true
+reorder_imports = true
+reorder_impl_items = true
+group_imports = "StdExternalCrate"
+use_try_shorthand = true
+wrap_comments = true
+max_width = 100
+overflow_delimited_expr = true
+remove_nested_parens = true
+reorder_modules = true
```
Comment (vkgnosis) [inline crates/driver/rustfmt.toml:17]: already the default

File: crates/driver/rustfmt.toml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+unstable_features = true
+use_field_init_shorthand = true
```
Comment (vkgnosis) [inline crates/driver/rustfmt.toml:19]: that's already a clippy warning but I guesse we might as well let rustfmt do it automatically

**Review by vkgnosis — APPROVED** [review]
Went through the options in the docs https://rust-lang.github.io/rustfmt/?version=v1.5.1&search= and left some comments.

We can remove the ones that just set a value that is already the default.

I'm fine with all the changes but for some of them there might be usability issues which is why they are unstable. For example one of the `use` related settings has an issue where rustfmt will silently delete comments the programmer has left between imports. Not necessarily a problem for us, just wanted to mention.

File: crates/driver/src/api/solve.rs [hunk 6]
```diff
@@ -1,12 +1,18 @@
-use crate::driver::Driver;
-use anyhow::Result;
-use model::auction::AuctionWithId;
-use shared::api::{
-    convert_json_response, error, extract_payload_with_max_size, ApiReply, IntoWarpReply,
+use {
```
Comment (nlordell) [inline crates/driver/src/api/solve.rs:1]: I actually think this looks aesthetically more pleasing. TBH, I didn't know you can write imports this way.

**Review by nlordell — APPROVED** [review]
I don't feel strongly about formatting as long as it is consistent everywhere (that being said, I subjectively like how the imports look now 😄).

Agree with @vkgnosis that I would prefer consistent formatting across the whole repo - so I would propose moving this to the root `rustfmt.toml` (can be done in a separate PR though). Don't know how others feel.

**Review by MartinquaXD — APPROVED** [review]
I also prefer the current grouping of imports but agree that ultimately it's only important that it's consistent and automated.

**Pull request merged**